### ISSUE TITLE
Digger produces many dead-end corridors

### ISSUE THREAD

**Issue by airstruck** [paulofmandown/rotLove#41]
Compare the rotLove digger example to [rot.js examples](http://ondras.github.io/rot.js/manual/#map/dungeon), which don't seem to produce any dead-ends.

Haven't looked into it closely yet, but it looks like maybe corridors are being dug to rooms that never get created because there isn't enough space.

**Comment by airstruck** [comment]
After reading up on the algorithm, I suspect priority walls aren't (always) getting created at the end of corridors as they should be. I found one bug in `Corridor:createPriorityWalls`:

```lua
if dx>0 then dx=dx/math.abs(dx) end
if dy>0 then dy=dy/math.abs(dy) end
```

Should be: 

```lua
if dx~=0 then dx=dx/math.abs(dx) end
if dy~=0 then dy=dy/math.abs(dy) end
```

The original JS:

```js
if (dx) { dx = dx/Math.abs(dx); }
if (dy) { dy = dy/Math.abs(dy); }
```

Unfortunately the dead-ends are still there after fixing that. I'm stumped for now.

**Comment by airstruck** [comment]
Found it, was one more instance like the one mentioned above. Fix in #41.
